### DISCUSSION TITLE
Postal services are up but can't access it through web

### DISCUSSION THREAD

**Discussion by nithinkumarmav** [postalserver/postal#3236] [Help with using Postal]
Postal seems to have installed correctly and is up and running:

# postal status
NAME                IMAGE                               COMMAND                  SERVICE             CREATED             STATUS              PORTS
postal-smtp-1       ghcr.io/postalserver/postal:3.3.4   "/docker-entrypoint.…"   smtp                21 minutes ago      Up 17 minutes       
postal-web-1        ghcr.io/postalserver/postal:3.3.4   "/docker-entrypoint.…"   web                 21 minutes ago      Up 17 minutes       
postal-worker-1     ghcr.io/postalserver/postal:3.3.4   "/docker-entrypoint.…"   worker              21 minutes ago      Up 17 minutes       

ufw is also set to allow incoming traffic to port 5000

ufw status
Status: active

To                         Action      From
--                         ------      ----
5000                       ALLOW       Anywhere                  
80/tcp                     ALLOW       Anywhere                  
443/tcp                    ALLOW       Anywhere                  
25/tcp                     ALLOW       Anywhere                  
2525/tcp                   ALLOW       Anywhere                  
587/tcp                    ALLOW       Anywhere                  
25,80,443,587,993/tcp      ALLOW       Anywhere                  
OpenSSH                    ALLOW       Anywhere                  
5000 (v6)                  ALLOW       Anywhere (v6)             
80/tcp (v6)                ALLOW       Anywhere (v6)             
443/tcp (v6)               ALLOW       Anywhere (v6)             
25/tcp (v6)                ALLOW       Anywhere (v6)             
2525/tcp (v6)              ALLOW       Anywhere (v6)             
587/tcp (v6)               ALLOW       Anywhere (v6)             
25,80,443,587,993/tcp (v6) ALLOW       Anywhere (v6)             
OpenSSH (v6)               ALLOW       Anywhere (v6)             

seems to be actually listening port 5000,but can't access it through web. here's the netstat output:

netstat -tulnp
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:5000            0.0.0.0:*               LISTEN      9901/puma 6.4.2 (tc 
tcp        0      0 127.0.0.54:53           0.0.0.0:*               LISTEN      459/systemd-resolve 
tcp        0      0 127.0.0.1:9090          0.0.0.0:*               LISTEN      9884/ruby           
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      459/systemd-resolve 
tcp        0      0 127.0.0.1:9091          0.0.0.0:*               LISTEN      9900/ruby           
tcp        0      0 127.0.0.1:2019          0.0.0.0:*               LISTEN      4515/caddy          
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      1/init              
tcp        0      0 127.0.0.1:3306          0.0.0.0:*               LISTEN      3570/docker-proxy   
tcp6       0      0 :::443                  :::*                    LISTEN      4515/caddy          
tcp6       0      0 :::25                   :::*                    LISTEN      9900/ruby           
tcp6       0      0 :::22                   :::*                    LISTEN      1/init              
tcp6       0      0 :::80                   :::*                    LISTEN      4515/caddy          
udp        0      0 127.0.0.54:53           0.0.0.0:*                           459/systemd-resolve 
udp        0      0 127.0.0.53:53           0.0.0.0:*                           459/systemd-resolve 
udp6       0      0 :::443                  :::*                                4515/caddy  
Here's the docker situation:

docker ps
CONTAINER ID   IMAGE                               COMMAND                  CREATED          STATUS          PORTS                      NAMES
5be4abab2f36   ghcr.io/postalserver/postal:3.3.4   "/docker-entrypoint.…"   32 minutes ago   Up 28 minutes                              postal-smtp-1
b9b88f393cc1   ghcr.io/postalserver/postal:3.3.4   "/docker-entrypoint.…"   32 minutes ago   Up 28 minutes                              postal-worker-1
bd50559d7d7e   ghcr.io/postalserver/postal:3.3.4   "/docker-entrypoint.…"   32 minutes ago   Up 28 minutes                              postal-web-1
045a210f5667   caddy                               "caddy run --config …"   2 hours ago      Up 2 hours                                 postal-caddy
c6e5606568bd   mariadb                             "docker-entrypoint.s…"   2 hours ago      Up 2 hours      127.0.0.1:3306->3306/tcp   postal-mariadb

It is showing the html when i command curl -v http://mysubdomain.mydomain.com:5000

> User-Agent: curl/8.9.1
> Accept: */*
> 
* Request completely sent off
< HTTP/1.1 200 OK

<!DOCTYPE html>
<html class='subPage'>
<head>
<title>Login - Postal</title>
<meta name="csrf-param" content="authenticity_token" />
<meta name="csrf-token" content="PHVrP3KzSabXfeL5h2QpWtfUIkwo9RL7n_rIK4ugL-rajeleHUtVfAwW5bP5mR6qfDABCiknsX1wsIOWVyq0Lg" />
<link rel="stylesheet" href="/assets/application/application-d582f69de04a4a1ddabf46f496aea06e97cee7e2d7e29b856091c57dfc9ec98d.css" data-turbolinks-track="reload" />
<script src="/assets/application/application-0dd293ea0a8f7f37880ae5bea4a4f1627195bd86b17430b7fb0313790f195df8.js" data-turbolinks-track="reload"></script>
<link href='/assets/favicon-cbf06fae3095e2043b79ae5eaa186df6c4dbaa0a2e5d6fef88baae67cc78fd79.png' rel='shortcut icon'>
<meta name="turbolinks-cache-control" content="no-cache">
</head>
<body>
<div class='subPageBox'>
<div class='subPageBox__title'>
Welcome to Postal
</div>

<div class='subPageBox__content'>
<form class="loginForm" action="/login" accept-charset="UTF-8" method="post"><input type="hidden" name="authenticity_token" value="sHoZ9mEfDMvd6YCv1XmlfThv_Al1AwEgtXryzVcHWlW3anWoq9zYJLE32uIZjFKBm0zIuWIIj1xB_IZKcaJLLg" autocomplete="off" /><input type="hidden" name="return_to" id="return_to" autocomplete="off" />
<p class='loginForm__input'><input type="email" name="email_address" id="email_address" value="" spellcheck="false" class="input input--text input--onWhite" placeholder="Your e-mail address" autofocus="autofocus" tabindex="1" /></p>
<p class='loginForm__input'><input type="password" name="password" id="password" value="" class="input input--text input--onWhite" placeholder="Your password" tabindex="2" /></p>
<div class='loginForm__submit'>
<ul class='loginForm__links'>
<li><a href="/login/reset">Forgotten your password?</a></li>
</ul>
<p><input type="submit" name="commit" value="Login" class="button button--positive" tabindex="3" data-disable-with="Login" /></p>
</div>
</form></div>

</div>
</body>
</html>

but I am not able to get to postal through http://mysubdomain.mydomain.com:5000/ through my browser.
And I just can't figure out what's wrong here, any pointers?

**Comment by willpower232** [comment]
You would normally use caddy or nginx or similar rather than accessing port 5000 directly, I expect that is the issue here as newer Postal/ruby on rails is a bit more sensitive to the domains accessing it.

**Reply by nithinkumarmav** [reply]
im using caddy by this command docker run -d \
   --name postal-caddy \
   --restart always \
   --network host \
   -v /opt/postal/config/Caddyfile:/etc/caddy/Caddyfile \
   -v /opt/postal/caddy-data:/data \
   caddy

**Reply by nithinkumarmav** [reply]
is there anything to be configured specifically for this ?(domain accessing)

**Reply by nithinkumarmav** [reply]
im new to postal, can you please help me with document(steps) nginx or similar

**Reply by willpower232** [reply]
what does your caddyfile look like? can you see caddy in `docker ps -a`?

**Reply by nithinkumarmav** [reply]
subdomain.domain.com {
  reverse_proxy 0.0.0.0:5000
}

this is the caddyfile, just modified domain & port while in setup

yeah able see in docker 

CONTAINER ID   IMAGE                               COMMAND                  CREATED       STATUS       PORTS                      NAMES
5be4abab2f36   ghcr.io/postalserver/postal:3.3.4   "/docker-entrypoint.…"   4 hours ago   Up 4 hours                              postal-smtp-1
b9b88f393cc1   ghcr.io/postalserver/postal:3.3.4   "/docker-entrypoint.…"   4 hours ago   Up 4 hours                              postal-worker-1

bd50559d7d7e   ghcr.io/postalserver/postal:3.3.4   "/docker-entrypoint.…"   4 hours ago   Up 4 hours                              postal-web-1

045a210f5667   caddy                               "caddy run --config …"   5 hours ago   Up 5 hours                              postal-caddy

c6e5606568bd   mariadb                             "docker-entrypoint.s…"   6 hours ago   Up 6 hours   127.0.0.1:3306->3306/tcp   postal-mariadb

**Reply by nithinkumarmav** [reply]
Can we connect quickly for 10 mins,if let me know the connectivity channel

**Reply by willpower232** [reply]
the reverse proxy should probably be 127.0.0.1 not 0.0.0.0 and then subdomain.domain.com should work for accessing Postals UI as long as subdomain.domain.com is what you've put in Postals config

**Reply by nithinkumarmav** [reply]
I have changed that but still the same , if possible can we connect

**Reply by nithinkumarmav** [reply]
Or, if you need any configuration file details to better understand my problem, please let me know

**Reply by willpower232** [reply]
what do you see in your browser when you go to https://mysubdomain.mydomain.com (note the https and no port)

**Reply by nithinkumarmav** [reply]
This site can’t be reached
DNS_PROBE_FINISHED_NXDOMAIN

**Reply by willpower232** [reply]
that means you haven't set up the DNS for it so is a completely different problem :sweat_smile: you can use https://whatsmydns.net to double check what the world is seeing for that record

**Reply by nithinkumarmav** [reply]
Actually, I just created a subdomain today, so that might have been the problem. Thank you so much for your help; it was really great for me. After the DNS configuration, if it works or if there are any other issues, I’ll let you know.

**Reply by nithinkumarmav** [reply]
Also I’m setting up Postal on a VPS(digital ocean) and want to ensure optimal email deliverability. Could you please share:

1.Which VPS providers are best suited for running Postal, considering email deliverability?
2.Any best practices or configurations to minimize emails being flagged as spam when using a VPS with Postal?

Please help me this as well, so that it will not cause a problem

**Reply by willpower232** [reply]
They all have their upsides and downsides, don't really recommend one over the other. If you've found one which will allow you to send emails then you can stick with them as they're apparently quite rare these days.

If you make sure you have all the ticks on the screen when you add a domain to Postal, then you can use things like https://mail-tester.com and https://mxtoolbox.com to confirm the messages you're sending are as good as they can be.

**Reply by nithinkumarmav** [reply]
Thanks @willpower232 .
How about Katapult? May I ask if you have used Katapult before? if so, how was your experience with it?

**Reply by willpower232** [reply]
I haven't used it personally but I know the developers of Postal work _closely_ with them so you'd hope Postal worked as well on there as it did anywhere else

**Comment by nithinkumarmav** [comment]
Hi, just following up on my question about setting up Postal on a VPS and ensuring good email deliverability. Would appreciate any advice. Thanks

**Reply by nithinkumarmav** [reply]
@willpower232  Could you please help to know the best suitable VPS for postal for US  and UK market. We keep researching for different VPS but finding difficult to identify the best fit for postal,Objective is to reduce spam to minimum.
Since we are in the critical path of the project your guidance will be timely and help.
Thanks in advance.
